cmake/xacc-config-version.cmake.in

```diff
@@ -0,0 +1,8 @@
+set(PACKAGE_VERSION "@MAJOR_VERSION@.@MINOR_VERSION@.@PATCH_VERSION@")
+# The quotes for the right operand are intentional here: if CMake executes this
+# file without ${PACKAGE_FIND_VERSION} set, then if() will still get three
+# operands. Without the quotes, if() will only get two operands in that case,
+# throwing an error.
+if(${PACKAGE_VERSION} VERSION_GREATER_EQUAL "${PACKAGE_FIND_VERSION}")
```

Comment:
I'm not sure if this behavior is right. For reference, here is the `LLVMConfigVersion.cmake` on my system:
```cmake
set(PACKAGE_VERSION "18.1.6")

# LLVM is API-compatible only with matching major.minor versions
# and patch versions not less than that requested.
if("18.1" VERSION_EQUAL
    "${PACKAGE_FIND_VERSION_MAJOR}.${PACKAGE_FIND_VERSION_MINOR}"
   AND NOT "6" VERSION_LESS "${PACKAGE_FIND_VERSION_PATCH}")
  set(PACKAGE_VERSION_COMPATIBLE 1)
  if("6" VERSION_EQUAL
      "${PACKAGE_FIND_VERSION_PATCH}")
    set(PACKAGE_VERSION_EXACT 1)
  endif()
endif()
```
Note that it's doing more careful logic there. If we release XACC 2.0 that breaks everything, then maybe this should not pass (it currently would say "ok, looks compatible to me")